app/models/scoped_event_feed.rb

```diff
@@ -0,0 +1,38 @@
+class ScopedEventFeed
+  attr_reader :root, :event_subjects, :page, :per
+  def initialize(root_type, root_id, page = 0, per = 20)
+    @root = root_type.classify.constantize.find(root_id)
+    @event_subjects = EventHierarchy.new(root).children
+    @page = page
+    @per = per
+  end
+
+  def events
+    if @events
+      @events
+    else
+      event_ids = event_subjects.group_by { |x| x.class.to_s }.inject([]) do |ids, (type, vals)|
```

Comment:
We get a list of `Event` subjects from our hierarchy, but unfortunately, the "or" methods available in ActiveRecord don't seem to play nicely with spanning multiple types in our polymorphic foreign key. (Ie: I can't just do Event.where(subject: subjects)` as it doesn't generate the SQL we'd want. 

Instead we group by the class of the subject (`Gene`, `Variant`, etc) and run one query for each type, but just pull back the Event `id` rather than the full object. So we'll have one query in the form of `select events.id from events where subject_type = 'Variant' and subject_id in (1, 2, 3);` for each type.

We aggregate these ids into a list of event ids and then perform our "real" query below, pulling back all the data we want, paginating, etc.